syntax = "proto3";
package workflow;

message WorkflowNode {
  // 工作流节点名称
  string name = 1;
  // 依赖(确认)
  repeated string dependencies = 2;
  // image
  string template = 3;
  // phase（状态）
  string phase = 4;
  // status
  string node_info = 5;
  // cpu(核数)
  uint32 cpu = 6;
  // 内存(Bytes)
  uint64 mem = 7;
  // 预计执行时间(单位：秒)
  uint32 duration = 8;
  // 环境变量
  map<string, string> env = 9;
  // 输入向量
  repeated string input_vector = 10;
  // 输出向量
  repeated string output_vector = 11;
}

message Workflow {
  // 工作流名称
  string workflow_name = 1;
  // 工作流custom_id
  string custom_id = 2;
  // 工作流类型
  string style = 3;
  // 是否为定制工作流
  bool customization = 4;
  // 时间等级(A,B,C,D)
  string time_grade = 5;
  // 花费（资金）等级(A,B,C,D)
  string cost_grade = 6;
  // 拓扑结构
  repeated WorkflowNode topology = 7;
}